store/db_key_layout.go

```diff
@@ -24,32 +25,40 @@ type v1LegacyLayout struct{}
 
 // CalcBlockCommitKey implements BlockKeyLayout.
 func (*v1LegacyLayout) CalcBlockCommitKey(height int64) []byte {
-	return []byte(fmt.Sprintf("C:%v", height))
+	return []byte("C:" + strconv.FormatInt(height, 10))
 }
 
 // CalcBlockHashKey implements BlockKeyLayout.
 func (*v1LegacyLayout) CalcBlockHashKey(hash []byte) []byte {
-	return []byte(fmt.Sprintf("BH:%x", hash))
+	hexHash := make([]byte, hex.EncodedLen(len(hash)))
+	hex.Encode(hexHash, hash)
+
+	return append([]byte{'B', 'H', ':'}, hexHash...)
 }
```

Comment:
Yes, this is faster than using `fmt.Sprintf` 😛 

On another note: why do we encode to hex the hash before using it as the key's suffix?